### PR TITLE
[ISSUE #32] add check_message_type to publishing setting.

### PR DESCRIPTION
**Pull request by lollipopjin**
add check_message_type to publishing setting,  no need to check message's type when this flag set false from server.

### CODE REVIEW

File: apache/rocketmq/v2/service.proto [hunk 1]
```diff
@@ -193,6 +193,10 @@ message Publishing {
   // reject the request. As a result, it is advisable that Producer performs
   // client-side check validation.
   int32 max_body_size = 3;
+
+  // When `check_message_type` flag set `false`, no need to check message's type
+  // with messageQueue's `accept_message_types` before publising.
+  optional bool check_message_type = 4;
```
Comment (aaron-ai) [inline apache/rocketmq/v2/service.proto:199]: When do we not fill the field since it is optional?

Comment (aaron-ai) [inline apache/rocketmq/v2/service.proto:199]: When do we not fill the field since it is optional?

Comment (lollipopjin) [reply apache/rocketmq/v2/service.proto]: fixed. anyway, the server will push it's default setting.

File: apache/rocketmq/v2/service.proto [hunk 2]
```diff
@@ -193,6 +193,10 @@ message Publishing {
   // reject the request. As a result, it is advisable that Producer performs
   // client-side check validation.
   int32 max_body_size = 3;
+
+  // When `check_message_type` flag set `false`, no need to check message's type
+  // with messageQueue's `accept_message_types` before publising.
+  bool check_message_type = 4;
```
Comment (lizhanhui) [inline apache/rocketmq/v2/service.proto:199]: It's generally discouraged to use bool in the protocol definition. Prefer explicit enum.
For example
enum ValidateMessageTypeStrategy {
 UNSPECIFIED = 0,
DISABLED = 1,
PERMISSIVE = 2,
ENFORCING = 3,
}

**Pull request merged**